core/templates/domain/topic/topic-rights-backend-api.service.ts

```diff
@@ -16,162 +16,188 @@
  * @fileoverview Service to change the rights of topic in the backend.
  */
 
-require('domain/utilities/url-interpolation.service.ts');
-
-require('domain/topic/topic-domain.constants.ajs.ts');
-
-angular.module('oppia').factory('TopicRightsBackendApiService', [
-  '$http', '$q', 'UrlInterpolationService',
-  'TOPIC_RIGHTS_URL_TEMPLATE',
-  function($http, $q, UrlInterpolationService,
-      TOPIC_RIGHTS_URL_TEMPLATE) {
-    // Maps previously loaded topic rights to their IDs.
-    var topicRightsCache = {};
-
-    var _fetchTopicRights = function(topicId, successCallback,
-        errorCallback) {
-      var topicRightsUrl = UrlInterpolationService.interpolateUrl(
-        TOPIC_RIGHTS_URL_TEMPLATE, {
-          topic_id: topicId
-        });
-
-      $http.get(topicRightsUrl).then(function(response) {
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { HttpClient } from '@angular/common/http';
+
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+
+import { TopicDomainConstants } from
+  'domain/topic/topic-domain.constants';
+import cloneDeep from 'lodash/cloneDeep';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TopicRightsBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private urlInterpolation: UrlInterpolationService) {}
+  // Maps previously loaded topic rights to their IDs.
+  private topicRightsCache = {};
+
+  private _fetchTopicRights(
+      topicId: string,
+      successCallback: (value?: Object | PromiseLike<Object>) => void,
+      errorCallback: (reason?: any) => void): void {
+    var topicRightsUrl = this.urlInterpolation.interpolateUrl(
+      TopicDomainConstants.TOPIC_RIGHTS_URL_TEMPLATE, {
+        topic_id: topicId
+      });
+
+    this.http.get(
+      topicRightsUrl, { observe: 'response' }).toPromise().then(
+      (response) => {
         if (successCallback) {
-          successCallback(response.data);
+          successCallback(response.body);
```

Comment:
Don't return the response from backend directly. Use a frontend object or create one if it doesn't exist. See #8015 for more details!

Reply:
Done